com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/security/data/termination/DataTerminationServiceTest.java

```diff
@@ -0,0 +1,181 @@
+        when(configuration.getGlobalAuditorPassword()).thenReturn(GLOBAL_AUDITOR_PASSWORD);
+    }
+
+    private List<String> generateDeletedSpaces(int countOfDeletedSpaces) {
+        List<String> spaces = new ArrayList<>();
+        for (int i = 0; i < countOfDeletedSpaces; i++) {
+            spaces.add(MessageFormat.format("space-{0}", i));
+        }
+        return spaces;
+    }
+
+    private void prepareCfOptimizedEventsGetter(List<String> deletedSpaces) {
+        when(cfOptmizedEventsGetter.findEvents(anyString(), anyString())).thenReturn(deletedSpaces);
+    }
+
+    private void prepareDaos(List<String> deletedSpaces, boolean isExistSubscriptionData, boolean isExistConfigurationEntryData) {
+        List<ConfigurationSubscription> subscriptions = generateSubscriptions(isExistSubscriptionData);
+        List<ConfigurationEntry> configurationEntries = generatedConfigurationEntries(isExistConfigurationEntryData);
+        deletedSpaces.forEach(deletedSpace -> initializeDaoMocks(subscriptions, configurationEntries, deletedSpace));
+    }
+
+    private List<ConfigurationSubscription> generateSubscriptions(boolean isExistSubscriptionData) {
+        return isExistSubscriptionData ? Arrays.asList(new ConfigurationSubscription()) : Collections.emptyList();
+    }
+
+    private List<ConfigurationEntry> generatedConfigurationEntries(boolean isExistConfigurationEntryData) {
+        return isExistConfigurationEntryData
+            ? Arrays.asList(new ConfigurationEntry("", "", Version.parseVersion("1"), new CloudTarget(), "", Collections.emptyList(), ""))
+            : Collections.emptyList();
+    }
+
+    private void initializeDaoMocks(List<ConfigurationSubscription> subscriptions, List<ConfigurationEntry> configurationEntries,
+                                    String deletedSpace) {
+        when(subscriptionDao.findAll(null, null, deletedSpace, null)).thenReturn(subscriptions);
+        when(entryDao.find(deletedSpace)).thenReturn(configurationEntries);
+    }
+
+    private void verifyDeleteOrphanUserData(int countOfDeletedSpaces, boolean isExistSubscriptionData,
+                                            boolean isExistConfigurationEntryData)
+        throws FileStorageException {
+        verify(fileService, atLeast(countOfDeletedSpaces)).deleteBySpace(anyString());
+
+        if (isExistSubscriptionData) {
+            verify(subscriptionDao, atLeast(countOfDeletedSpaces)).removeAll(any());
+        } else {
+            verify(subscriptionDao, never()).removeAll(any());
+        }
+
+        if (isExistConfigurationEntryData) {
+            verify(entryDao, atLeast(countOfDeletedSpaces)).removeAll(any());
+        } else {
+            verify(entryDao, never()).removeAll(any());
+        }
+
+    }
+
+    @Test
+    public void testMissingAuditorCredentials() {
+        assertThrows(IllegalStateException.class, () -> dataTerminationService.deleteOrphanUserData(),
```

Comment:
If you're trying to check whether the message which exception has thrown equals to the message Messages.MISSING_GLOBAL_AUDITOR_CREDENTIALS, you have to do sth like this:
Exception exception = assertThrows(SOME_EXCEPTION.class, () -> ....);
assertEquals(EXPECTED_MESSAGE, exception.getMessage());